rewrite-maven/src/main/java/org/openrewrite/maven/utilities/MavenArtifactHelper.java

```diff
@@ -0,0 +1,126 @@
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven.utilities;
+
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.maven.MavenExecutionContextView;
+import org.openrewrite.maven.cache.InMemoryMavenPomCache;
+import org.openrewrite.maven.cache.LocalMavenArtifactCache;
+import org.openrewrite.maven.cache.ReadOnlyLocalMavenArtifactCache;
+import org.openrewrite.maven.internal.MavenPomDownloader;
+import org.openrewrite.maven.internal.RawMaven;
+import org.openrewrite.maven.internal.RawMavenResolver;
+import org.openrewrite.maven.tree.*;
+import org.openrewrite.xml.tree.Xml;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.function.Predicate;
+
+public class MavenArtifactHelper {
+
+    private static final MavenRepository SUPER_POM_REPOSITORY = new MavenRepository("central",
+            URI.create("https://repo.maven.apache.org/maven2"), true, false, null, null);
+
+    public static List<Path> downloadArtifactAndDependencies(String groupId, String artifactId, String version, ExecutionContext ctx) {
+        List<MavenRepository> repositories = new ArrayList<>();
+        repositories.add(SUPER_POM_REPOSITORY);
+        return downloadArtifactAndDependencies(groupId, artifactId, version, ctx, repositories);
+    }
+
+    public static List<Path> downloadArtifactAndDependencies(String groupId, String artifactId, String version, ExecutionContext ctx, List<MavenRepository> repositories) {
+        MavenPomDownloader mavenPomDownloader = new MavenPomDownloader(new InMemoryMavenPomCache(),
+                Collections.emptyMap(), ctx);
+        RawMaven rawMaven = mavenPomDownloader.download(groupId, artifactId, version, null, null,
+                repositories, ctx);
+        if (rawMaven == null) {
+            return Collections.emptyList();
+        }
+        MavenExecutionContextView mavenCtx = new MavenExecutionContextView(ctx);
+        mavenCtx.setRepositories(repositories);
+        Xml.Document xml = new RawMavenResolver(mavenPomDownloader, Collections.emptyList(), true, ctx, null).resolve(rawMaven, new HashMap<>());
+        if (xml == null) {
+            return Collections.emptyList();
+        }
+        Maven maven = new Maven(xml);
+        Pom pom = maven.getModel();
+        MavenArtifactDownloader mavenArtifactDownloader = new MavenArtifactDownloader(ReadOnlyLocalMavenArtifactCache.MAVEN_LOCAL.orElse(
+                new LocalMavenArtifactCache(Paths.get(System.getProperty("user.home"), ".rewrite-cache", "artifacts"))
```

Comment:
I think Aaron mentioned this already but we probably should change this to `~/.rewrite/cache/artifacts`